src/utils.rs

```diff
@@ -212,3 +212,28 @@ pub fn format_pairing_check_values(
 
     Ok([(msg_hash_arr, pk_arr), (sig_arr, n_g2)])
 }
+
+pub fn format_pairing_check_uncompressed_values(
+    message: Vec<u8>,
+    mut signature: Vec<u8>,
+    mut public_key: Vec<u8>,
+) -> Result<[([u8; 64], [u8; 128]); 2]> {
+    // convert to little endian
+    for i in [0, 32] {
+        signature[i..i + 32].reverse()
+    }
+    for i in [0, 32, 64, 96] {
```

Comment:
More rust like to do:

`for x in (0..=96).step_by(32)`. 
 Same for one above, but it's corresponding numbers.